crates/transcode/src/lib.rs

```diff
@@ -184,6 +178,13 @@ impl ContractMessageTranscoder {
         Ok(Self::new(ink_metadata))
     }
 
+    /// Constructs an instance of `Self` out of a contract metadata.
+    ///
+    /// Returns error if can't deserialize metadata's ABI.
+    pub fn from_metadata(metadata: contract_metadata::ContractMetadata) -> Result<Self> {
```

Comment:
This could be `impl TryFrom<ContractMetadata>`

Reply:
Done in [`e0d3e79` (#804)](https://github.com/paritytech/cargo-contract/pull/804/commits/e0d3e795be6f2b6fedc6ee173b263db2f35b8127)